proposals/dgl-operator-proposal.md

```diff
@@ -0,0 +1,284 @@
+## Motivation
+
+> Deep Graph Library ([DGL](https://github.com/dmlc/dgl)) is a Python package built for easy implementation of graph neural network model family, on top of existing DL frameworks (currently supporting PyTorch, MXNet and TensorFlow). It offers a versatile control of message passing, speed optimization via auto-batching and highly tuned sparse matrix kernels, and multi-GPU/CPU training to scale to graphs of hundreds of millions of nodes and edges. —— DGL team @ [dgl.ai](https://www.dgl.ai/)
+
+More recently, there are a lot of new deep learning researches based on graph-structured data. And DGL has become one of the best GNN frameworks that researcher used to fulfill their works. The researchers leverage DGL to solve some conventional graph related problems, like social networks, chemical molecules and recommender systems. The DGL eco-system also has many high-leveled packages, which are developed for domain applications, such as [DGL-KE](https://github.com/awslabs/dgl-ke), [DGL-LifeSci](https://github.com/awslabs/dgl-lifesci) and DGL-RecSys. In Qihoo 360, some ML/DL teams would like to take experiments with DGL and DGL-KE framework, so we just kicked off the project on distributed DGL training.
+
+[Native DGL distributed training](https://docs.dgl.ai/guide/distributed.html) is based on data parallelism, and we (360 AI Infra team) applied this fully distributed approach on operator/controller philosophy for Kubernetes. This proposal is aimed at defining what the DGL Operator should behave, and add it to Kubeflow.
+
+## Goals
+
+A Kubeflow user should be able to run training using DGL as easily as then can using Tensorflow or PyTorch. This proposal is centered around a Kubernetes operator for DGL. A user should be able to run both single worker job with DGL (1 launcher and 1 worker) and distributed training jobs with DGL (1 launcher and multiple workers).
+
+This proposal defines the following:
+
+- A DGL operator
+- A way to deploy the DGL Operator
+- A single worker DGL example
+- A distributed (multiple workers) DGL example
+
+## Non-Goals
+
+Currently, for the scope of this proposal, we won't be addressing the method for serving the model.
+
+## API
+
+### Custom Resource Definition
+
+The CRD yaml example is as following:
+
+```yaml
+apiVersion: qihoo.net/v1alpha1
```

Comment:
I assume apiVersion will be changed to kubeflow fashion?

Reply:
Yes, apiVersion will be changed to kubeflow fashion